### PR TITLE
System debug: move to SBU from IOT Kit

### PR DESCRIPTION
**Pull request by daverous**
# Description

This moves the System Debug sample to the System Bring up toolkit away from the IOT Kit. 

While the SBU product will require an NDA, this sample will remain public. 

This also has some QOL improvements and refactors the repository with submodules correctly. 
Fixes # (issue) 

ONSAM-1323
ONSAM-1330
## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [x] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [x] Implement coding guidelines and ensure code quality.
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [x] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Enforce format via clang-format config file
- [x] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: Tools/SystemDebug/sample.json [hunk 1]
```diff
@@ -1,9 +1,8 @@
 {
   "guid": "668A3FF9-12C3-4559-8AF4-54B6792122D3",
   "name": "Intel System Debugger Tutorial (Overview)",
-  "categories": ["Toolkit/Intel® oneAPI IoT Toolkit/System Debugger"],
+  "categories": ["Toolkit/Intel® System Bring-Up Toolkit/System Debugger"],
   "description": "Demonstrate basic functionality of the Intel System Debugger.",
-  "os": ["linux", "windows"],
```
Comment (JoeOster) [inline Tools/SystemDebug/sample.json:6]: as this seems to be a power  shell script, the OS  should be windows

**Review by JoeOster — DISMISSED** [review]

**Comment by anjgola** [comment]
json validation is failing because of the capital u in system bring-up name. It needs to be small case.

File: Tools/SystemDebug/sample.json (hunk 1 above)
Comment (daverous) [reply Tools/SystemDebug/sample.json:6]: The powershell script is to flash the efi_application - I was told to remove the OS as it shouldn't show up in the samples browser in visual studio or eclipse

Comment (JoeOster) [reply Tools/SystemDebug/sample.json:6]: Thanks for the clarification

**Review by JoeOster — APPROVED** [review]

**Review by JoeOster — CHANGES_REQUESTED** [review]
@daverous - your sample is not passing ci, please investigate and resolve

**Review by JoeOster — DISMISSED** [review]

**Comment by JoeOster** [comment]
@anjgola - any idea why the windows validation is still pending?

**Comment by daverous** [comment]
Why are we still Pending on windows/darwin validation? 

Best,

David

**Comment by anjgola** [comment]
Ekaterina, Victor have to answer why the windows and darwin are showing pending.

**Comment by anjgola** [comment]
Failure on Mac due to submodules. But well, this shouldn't have run on Mac in the first place, I think when a PR is triggered manually it gets triggered on all OS.
Waiting for windows run to happen, keeps failing with 443 but I keep retriggering it, let us see who wins.

**Comment by anjgola** [comment]
CI team said they have fixed the credentials problem, re running.

**Review by JoeOster — APPROVED** [review]

**Pull request merged**